packages/react/src/components/UIShell/SideNav.js

```diff
@@ -121,14 +121,21 @@ const SideNav = React.forwardRef(function SideNav(props, ref) {
     eventHandlers.onMouseLeave = () => handleToggle(false, false);
   }
 
+  function getAriaHiddenValue() {
+    if (expanded === false && window.innerWidth < 1056) {
+      return true;
+    }
+    return false;
+  }
+
```

Comment:
I think we'll need to do two things for this:

- Make sure we don't call `window` if the component is being SSR'd
- Make sure that this value doesn't go "stale", in other words this component should update if the width goes below 1056 or greater than 1056 so we don't get a mismatch when the browser resizes

For the first part, we can use `src/internal/environment` to get `canUseDOM` to guard against calling `window`, we can also put calls to window in a `useEffect` which don't run on the server.

This comes up for the second case where we most likely want to use `window.matchMedia`: https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia

This is a browser API that lets us specify a media query and listen to changes for if the media query applies, or not. Here's a codesandbox to try and show this at play: https://codesandbox.io/s/amazing-kapitsa-6lyd5 👀

Reply:
ah yeah, good call. Vince did something [similar](https://github.com/carbon-design-system/gatsby-theme-carbon/blob/c9a8e074a459c65ede37aea578d48c266cce3bc4/packages/gatsby-theme-carbon/src/components/ArtDirection.js#L7-L8) for `ArtDirection` component of Gatsby theme, bc it changes w/ diff breakpoints.